### ISSUE TITLE
Issue with RTL languages in PDF invoices

### ISSUE THREAD

**Issue by RomanovM** [nopSolutions/nopCommerce#6614]
nopCommerce version: 4.60

> we have a problem with english text and dates, everything is upside down.

> Its very easy to reproduce,
> Create a product like: אוזניות אלחוטיות Xiaomi Redmi Buds 3 Lite - צבע שחור
> Create Shipping method name like: משלוח Post לנקודת איסוף הקרובה
> Try to make an order - and then print invoice PDF.
> 
> on version 4.5.* there was no such problem, after updating to version 4.6.1, this problem appeared.

Source: https://www.nopcommerce.com/boards/topic/96345/pdf-invoice-rtl-issue

The problem is when in one field there are both en characters and local language characters, in this case the en characters are inverted.

Related work item: https://github.com/nopSolutions/nopCommerce/issues/7477

**Comment by AndreiMaz** [comment]
Waiting for the fix in the original PDF library - https://github.com/QuestPDF/QuestPDF/issues/445

**Comment by navongol** [comment]
> Waiting for the fix in the original PDF library - [QuestPDF/QuestPDF#445](https://github.com/QuestPDF/QuestPDF/issues/445)

Please see his answer, are you going to wait for a solution?

**Comment by navongol** [comment]
> > Waiting for the fix in the original PDF library - [QuestPDF/QuestPDF#445](https://github.com/QuestPDF/QuestPDF/issues/445)
> 
> Please see his answer from today, are you going to wait for a solution?
> 
> ![image](https://user-images.githubusercontent.com/17260191/239694262-24f941fc-c8df-4585-9b60-20944335d413.png)

andrei hi,
what is the plan for this issue?
it's affect us, we are waiting for a solution.

thanks in advance

**Comment by AndreiMaz** [comment]
@navongol Yes, currently we're waiting for a solution from the PDF library

**Comment by A-Jawad** [comment]
Same Issue with numbers in RTL PDF invoice,
the numbers of price, date  appear in reverse direction:
30 Jan. 2023 appear as 03 Jan.3202
$1000.00  appear as $0001.00
so the admin and clients are unable to print a pdf.

**Comment by navongol** [comment]
It's been a long time, what's going on?

**Comment by navongol** [comment]
> @navongol Yes, currently we're waiting for a solution from the PDF library

It's been a long time, what's going on?

**Comment by hosamhemaily** [comment]
any one fix or make any work around

**Comment by rajupaladiya** [comment]
@RomanovM  / @AndreiMaz - Seems Quest PDF has fixed this issue,
Kindly check [this comment ](https://github.com/QuestPDF/QuestPDF/issues/445#issuecomment-1955960910)

**Comment by navongol** [comment]
![Screenshot 2024-04-17 121738](https://github.com/nopSolutions/nopCommerce/assets/17260191/b081b34f-e700-4d0a-9b08-1640a9a00e84)


When it will be fixed ?!

**Comment by navongol** [comment]
I am considering consult my customers move to other solution, leave nopcommerce ....

**Comment by rajupaladiya** [comment]
@AndreiMaz - is this fixed in nopCommerce 4.7 release?

**Comment by AndreiMaz** [comment]
@rajupaladiya No, this work item is still open

**Comment by arshyuo2** [comment]
There is a problem of incorrect display of numbers and some letters in version 4.7.01.
Most likely, this problem is due to the font used in the program for PDF output of orders
![order pdf file](https://github.com/nopSolutions/nopCommerce/assets/65789306/ef8e74d2-4193-47a8-b70e-e42143a7ad48)

**Comment by hossein2408** [comment]
pleas fix it

**Comment by hosamhemaily** [comment]
Not fixed

**Comment by rajupaladiya** [comment]
Any update please? are we going to fix this or planning to use back iTextSharp library?

**Comment by AndreiMaz** [comment]
Related work item: https://github.com/nopSolutions/nopCommerce/issues/7477

**Comment by exileDev** [comment]
Well, it's closed in #7477. Please feel free to test and provide feedback. To use Hebrew you need to add TTF-font (~\App_Data\Pdf) and change the PdfSettings.RtlFontName setting accordingly.
